### PR TITLE
Added 2 modules

### PR DESCRIPTION
**Pull request by IzmaylovI**
1) Modified the timer class
2) The benchmark class has been finalized
3) The project has been tested for assembly and code style

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,5 +28,17 @@ else()
...
+# Link OpenCV
+find_package(OpenCV REQUIRED)
+if(OpenCV_FOUND)
+    message(STATUS "Link OpenCV")
+    include_directories( ${OpenCV_INCLUDE_DIRS} )
+    add_compile_definitions(WITH_OPENCV2)
+    target_link_libraries(computer_vision ${OpenCV_LIBS})
+else()
+    message(STATUS "Not link OpenCV")
+endif()
```
Comment (aobolensk) [inline CMakeLists.txt]: To test how library works without OpenCV, we actually need to link with OpenCV when specific flag is provided on cmake configuration step

Comment (IzmaylovI) [reply CMakeLists.txt]: Is it possible to test how library works without OpenCV if add `option` to `cmake`?

Comment (aobolensk) [reply CMakeLists.txt]: Sure, we can build it twice: first time with OpenCV and the second time without it. This looks like that:
```
cmake -DENABLE_OPENCV=ON ..
cmake --build ...
...
cd .. && rm -r build && mkdir build && cd build
cmake -DENABLE_OPENCV=OFF ..
cmake --build ...
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -46,4 +46,4 @@ jobs:
         path: |
           opencv
           build_opencv
-        key: ubuntu-x86-cross-build-opencv
+        key: ubuntu-x86-cross-build-opencv
```
Comment (aobolensk) [inline .github/workflows/ci.yml]: Please, revert line removal

File: algorithms/headers/function.hpp [hunk 3]
```diff
@@ -4,8 +4,8 @@
 #include <arm_neon.h>
 #include <omp.h>
 
-#define cllps 3
-#define thr 4
+#define cllps 1
+#define thr 1
```
Comment (aobolensk) [inline algorithms/headers/function.hpp:8]: Is it needed?

Comment (IzmaylovI) [reply algorithms/headers/function.hpp:8]: This is for the convenience of changing the number of threads and the number of parallel nested loops in matrix multiplication.

Comment (aobolensk) [reply algorithms/headers/function.hpp:8]: I mean this change from 3 and 4 to 1

Comment (IzmaylovI) [reply algorithms/headers/function.hpp:8]: There was no need to change 3 and 4 to 1 in this I guess.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (IzmaylovI) [reply .github/workflows/ci.yml]: Do you mean revert the commit?

Comment (aobolensk) [reply .github/workflows/ci.yml]: Cancel this change, because it is not needed

**Comment by allnes** [comment]
@IzmaylovI please resolve conflicts

**Comment by IzmaylovI** [comment]
Are there any unresolved conflicts?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (IzmaylovI) [reply .github/workflows/ci.yml]: Can you please tell me if the changes are correct?

Comment (aobolensk) [reply .github/workflows/ci.yml]: Remove line 49

Comment (IzmaylovI) [reply .github/workflows/ci.yml]: Deleted line 49

**Review by aobolensk — APPROVED** [review]

**Pull request merged**